src/player.js

```diff
@@ -1183,23 +1185,21 @@ export default class Player extends FakeEventTarget {
    * @returns {void}
    */
   notifyExitFullscreen(): void {
-    if (this._fullscreen) {
-      this._fullscreen = false;
+    if (!this.isFullscreen()) {
       this.dispatchEvent(new FakeEvent(CustomEventType.EXIT_FULLSCREEN));
     }
   }
 
   /**
    * Request the player to enter fullscreen.
    * @public
+   * @param {HTMLElement} fullScreenElement - set html element to full screen
    * @returns {void}
    */
-  enterFullscreen(): void {
-    if (!this._fullscreen) {
-      if (this.isInPictureInPicture()) {
-        this.exitPictureInPicture();
-      }
+  enterFullscreen(fullScreenElement: ?HTMLElement): void {
+    if (!this.isFullscreen()) {
```

Comment:
as you now have a controller any conditional logic and events should be handled there.
The API should be a simple proxy to the controller.